src/storage/useFilePreview.ts

```diff
@@ -0,0 +1,60 @@
+'use client'
+
+import { useQuery, useQueryClient, UseQueryOptions } from '@tanstack/react-query'
+import { useMemo } from 'react'
+import { useAppwrite } from 'react-appwrite'
+
+type ImageOptionsProp = {
+  width?: number,
+  height?: number,
+  gravity?: number,
+  quality?: number,
+  borderWidth?: number,
+  borderColor?: string,
+  borderRadius?: number,
+  opacity?: number,
+  rotation?: number,
+  background?: string,
+  output?: string,
+}
+
+/**
+ * Access to the file object by its unique file ID.
+ * @param bucketId Storage bucket unique ID
+ * @param fileId  File ID
+ * @param imageOptions Options for cutting and resizing your preview image.
+ * @param options Options to pass to `react-query`
+ * @link [Appwrite Documentation](https://appwrite.io/docs/client/storage?sdk=web-default#storageGetFilePreview)
+ */
+export function useFilePreview(
+  bucketId: string,
+  fileId: string,
+  imageOptions: ImageOptionsProp = {
+    width: undefined,
+    height: undefined,
+    gravity: undefined,
+    quality: undefined,
+    borderWidth: undefined,
+    borderColor: undefined,
+    borderRadius: undefined,
+    opacity: undefined,
+    rotation: undefined,
+    background: undefined,
+    output: undefined,
+  },
```

Comment:
We should not have these undefineds here. It should essentially look similar to how `useAvatar` looks, but instead of `Avatar` type, we'd have a `Preview` type.

https://github.com/sanny-io/react-appwrite/blob/2a9af7d108d6e1340a830dc0b6f951e7cc9ed200/src/avatars/useAvatar.ts